src/utils/vectors.ts

```diff
@@ -0,0 +1,97 @@
+import { emptyCost, Generator, GeneratorInstance } from '../armature/Generator';
+import { Node } from '../armature/Node';
+
+import { vec4 } from 'gl-matrix';
+import { range } from 'lodash';
+
+const vectorPool: vec4[] = [];
+let nextVectorIndex: number = 0;
+const zero = vec4.fromValues(0, 0, 0, 1);
+
+// Convenience functions to reuse a pool of vectors
+const getNewVector = () => {
+    if (nextVectorIndex < vectorPool.length) {
+        const nextVector = vectorPool[nextVectorIndex];
+        nextVectorIndex += 1;
+
+        return nextVector;
+    } else {
+        const nextVector = vec4.create();
+        vectorPool.push(nextVector);
+        nextVectorIndex += 1;
+
+        return nextVector;
+    }
+};
+
+const done = () => {
+    nextVectorIndex = 0;
+};
+
+/**
+ * Given a generator and a component to generate, creates a set of direction vectors generated as
+ * children of that component.
```

Comment:
It's not clear what direction the vectors are pointing to from this description. Can you say something like "create a set of direction vectors that each point from a parent node to one of its child nodes"?